### ISSUE TITLE
Appearance color

### ISSUE THREAD

**Issue by philipengberg** [intercom/intercom-ios#145]
## Version info
  - Intercom for iOS version: `3.0.0`
  - iOS version: `9.3.2`

## Expected behavior
Consistent coloring

## Actual behavior
Variable coloring

## Steps to reproduce
I have set the Messenger color to `#00ffc8`, which is our app's primary color. Sure, it's a delicate color, but it should be possible to handle it dynamically.

First of all, the navigation bar of the Messages List has a super weird color:

![simulator screen shot 22 jul 2016 08 07 23](https://cloud.githubusercontent.com/assets/1840093/17048126/bfa21612-4fe3-11e6-8cb2-16153f1fd7bb.png)

which is not consistent with the Message view (which, by the way, is actually consistent with the general look of our app):

![simulator screen shot 22 jul 2016 08 07 29](https://cloud.githubusercontent.com/assets/1840093/17048131/d08b4818-4fe3-11e6-96a0-5a3b87798a49.png)

Second, as you can see, the messages are basically unreadable with the white text on top this bright teal color, however, that is **not** the case with the image placeholders on the Message List. Here the color has automatically been darkened.

## Suggestions

1. Make the appearance editor on the website more detailed, so that the color of different elements can be customized to look good.
2. Use one of multiple libraries to figure out if the chosen appearance color is bright or dark to make sure to pick a readable text color for the messages overview.

## Summary

To sum up, there are actually three issues:
1. Navigation bar appearance is not consistent
2. Message text color does not handle bright background color
3. Appearance color is not consistent across views

**Comment by Jonaswbu** [comment]
- Update to new Tonsser-icon.
- If we can't control the white text on buttons and icons, i suggest we use black instead of Tonsser-green.

**Comment by jtreanor** [comment]
Thanks @philipengberg for the detailed report.

The issues shown in your screenshots is that the messenger is applying UIAppearance proxies inconsistly. This version of the Messenger is not designed to be customised in this way, so we will release an update which is resilient against this. I'll keep you updated.

The other part of your post is a request for better customisation of the Messenger. We hear you and will absolutely take it on board.

**Comment by jtreanor** [comment]
Let me know if you have any questions :)

**Comment by jtreanor** [comment]
We have just released [3.0.1](https://github.com/intercom/intercom-ios/releases/tag/3.0.1) which includes the changes described above.

We have taken note of your request for further customization.

**Comment by philipengberg** [comment]
That was fast @jtreanor 👍

**Comment by phips28** [comment]
@jtreanor there is still one styling issue:

If I set: `[[UINavigationBar appearance] setBarStyle:UIBarStyleBlack];`
the messenger gets a grey navBar:
![simulator screen shot jul 22 2016 1 07 59 pm](https://cloud.githubusercontent.com/assets/3694800/17069960/557775fe-500e-11e6-9255-f8c70c6deea4.png)
![simulator screen shot jul 22 2016 1 08 02 pm](https://cloud.githubusercontent.com/assets/3694800/17069963/5ab6ca7e-500e-11e6-8d6e-3fdf28c4f352.png)

**Comment by jtreanor** [comment]
Thanks for the heads up on this one. I'll include it in the next release.

**Comment by jtreanor** [comment]
I'm reopening this until we get that final fix in. In the meantime, a workaround is either not to set barStyle using UIAppearance (and set it directly on your navigation bar) or to use something like this instead:

```
[[UINavigationBar appearanceWhenContainedIn:[MyWindow class]] setBarStyle:UIBarStyleBlack]
```

**Comment by jtreanor** [comment]
This is now fixed in [3.0.3](https://github.com/intercom/intercom-ios/releases/tag/3.0.3).
